### PR TITLE
Change license and remove references to PATENTS

### PR DESCRIPTION
**Pull request by sophiebits**
Only remaining references:

```
docs/_posts/2014-10-28-react-v0.12.md
51:You can read the full text of the [LICENSE](https://github.com/facebook/react/blob/master/LICENSE) and [`PATENTS`](https://github.com/facebook/react/blob/master/PATENTS) files on GitHub.

docs/_posts/2015-04-17-react-native-v0.4.md
20:* **Patent Grant**: Many of you had concerns and questions around the PATENTS file. We pushed [a new version of the grant](https://code.facebook.com/posts/1639473982937255/updating-our-open-source-patent-grant/).

src/__mocks__/vendor/third_party/WebComponents.js
8: * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
```

### CODE REVIEW

File: scripts/rollup/packaging.js [hunk 1]
```diff
@@ -128,10 +128,7 @@ function copyNodePackageTemplate(packageName) {
   // if the package directory already exists, we skip copying to it
   if (!fs.existsSync(to) && fs.existsSync(from)) {
     return asyncCopyTo(from, to).then(() =>
-      Promise.all([
-        asyncCopyTo(resolve('./LICENSE'), `${to}/LICENSE`),
-        asyncCopyTo(resolve('./PATENTS'), `${to}/PATENTS`),
-      ])
+      Promise.all([asyncCopyTo(resolve('./LICENSE'), `${to}/LICENSE`)])
```
Comment (gaearon) [inline scripts/rollup/packaging.js]: Seems like we can get rid of `Promise.all` here?

File: LICENSE [hunk 2]
```diff
@@ -1,31 +1,22 @@
-BSD License
+MIT License
 
 For React software
 
 Copyright (c) 2013-present, Facebook, Inc.
-All rights reserved.
```
Comment (frenzzy) [inline LICENSE]: This line was removed from LICENSE file, but not from other source files... :blush:

Comment (sophiebits) [reply LICENSE]: Thanks, fixed.

**Comment by frenzzy** [comment]
Do we still need `LICENSE-docs` and `LICENSE-examples` files in the root directory of the source tree?

**Pull request merged**